packages/rps/src/utils/channelClient.ts

```diff
@@ -0,0 +1,219 @@
+
+  /**
+   * Message payload. Format defined by wallet and opaque to app.
+   */
+  data: string;
+}
+
+export interface Funding {
+  token: string;
+  amount: string;
+}
+
+export interface ParameterHash {
+  [key: string]: any;
+}
+
+export interface JsonRPCNotification<ParametersType = ParameterHash, MethodType = string> {
+  jsonrpc: '2.0';
+  method: MethodType;
+  params: ParametersType;
+}
+
+export interface JsonRPCRequest<ParametersType = ParameterHash, MethodType = string>
+  extends JsonRPCNotification<ParametersType, MethodType> {
+  id: string;
+}
+
+export interface JsonRPCResponse<ResultType = {[key: string]: any}> {
+  jsonrpc: '2.0';
+  id: string;
+  result: ResultType;
+}
+
+export interface CreateChannelParameters extends UpdateChannelParameters {
+  appDefinition: string;
+}
+
+export interface ChannelResult extends CreateChannelParameters {
+  channelId: string;
+  status: ChannelStatus;
+  funding: Funding[];
+}
+
+export interface CreateChannelRequest
+  extends JsonRPCRequest<CreateChannelParameters, MethodName.CreateChannel> {}
+
+export interface CreateChannelResult extends ChannelResult {
+  turnNum: 0;
+}
+
+export interface CreateChannelResponse extends JsonRPCResponse<ChannelResult> {}
+
+export interface JoinChannelParameters {
+  channelId: string;
+}
+
+export interface JoinChannelRequest
+  extends JsonRPCRequest<JoinChannelParameters, MethodName.JoinChannel> {}
+
+export interface JoinChannelResult extends ChannelResult {
+  turnNum: 1;
+}
+
+export interface JoinChannelResponse extends JsonRPCResponse<ChannelResult> {}
+
+export interface UpdateChannelParameters {
+  participants: Participant[];
+  allocations: Allocation[];
+  appData: string;
+}
+
+export interface UpdateChannelRequest
+  extends JsonRPCRequest<UpdateChannelParameters, MethodName.UpdateChannel> {}
+
+export interface UpdateChannelResult extends ChannelResult {
+  turnNum: number;
+}
+
+export interface ChannelProposedNotification
+  extends JsonRPCNotification<CreateChannelResult, 'ChannelProposed'> {}
+
+export interface ChannelUpdatedNotification
+  extends JsonRPCNotification<UpdateChannelResult, 'ChannelUpdated'> {}
+
+export type ActionParameters =
+  | CreateChannelParameters
+  | JoinChannelParameters
+  | UpdateChannelParameters;
+
+export class ChannelClient {
+  protected events = new EventEmitter();
+
+  onMessageReceived(
+    notificationName: NotificationName,
+    callback: (parameters: ActionParameters) => void
+  ) {
+    this.events.on(
+      'message',
+      (message: JsonRPCNotification<ActionParameters> | JsonRPCRequest<ActionParameters>) => {
+        if (message.method === notificationName) {
+          callback(message.params);
+        }
+      }
+    );
+  }
+
+  createChannel(parameters: CreateChannelParameters) {
+    this.send(MethodName.CreateChannel, parameters);
+
+    // TODO: This notification payload is incomplete, since it doesn't have the result
+    // of the channel creation. Where should this go?

```

Comment:
For faking purposes, this code should add a `channelId`, `turnNum: 0`, `status: 'opening'` etc. to the hash. The `ChannelClient` will also need to store the last state of the channel internally, to allow it to fake out the result of other methods.

Once we stop faking, this method will probably need to wait for the response to the `CreateChannel` call from the wallet. This method (and others) will need to be `async`. We should probably make that change now.